### ISSUE TITLE
Build failed with w64devkit on windows

### ISSUE THREAD

**Issue by pingwuu** [MartinHelmut/cpp-gui-template-sdl2#19]
I try to build with w64devkit on windows, but failed while link final App.exe
Version Info:
Host OS:          Windows 10  version: 22H2 (19045.4046)
w64devkit:       w64devkit-1.21.0
Ninja:               1.11.1
Cmake:             cmake-3.28.3-windows-x86_64

Configure command:  > cmake -GNinja -DCMAKE_BUILD_TYPE=Debug -B build/debug2
Build command:          > cmake --build build/debug

Error output log:
=================================
>cmake --build build/debug2
......

[513/513] Linking CXX executable src\app\App.exe
FAILED: src/app/App.exe
C:\Windows\system32\cmd.exe /C "cd . && D:\w64devkit\bin\c++.exe -g /subsystem:windows /entry:mainCRTStartup  -mwindows src/app/CMakeFiles/App.dir/App/Main.cpp.obj src/app/CMakeFiles/App.dir/Manifests/app.rc.obj -o src\app\App.exe -Wl,--out-implib,src\app\libApp.dll.a -Wl,--major-image-version,0,--minor-image-version,0  src/core/libCore.a  _deps/spdlog-build/libspdlogd.a  _deps/fmt-build/libfmtd.a  vendor/imgui/libimgui.a  _deps/sdl2-build/libSDL2d.dll.a  src/settings/libSettings.a  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && C:\Windows\system32\cmd.exe /C "cd /D D:\20_gui\cpp-gui-template-sdl2\build\debug2\src\app && D:\cmake-3.28.3\bin\cmake.exe -E copy_if_different D:/20_gui/cpp-gui-template-sdl2/build/debug2/_deps/sdl2-build/SDL2d.dll D:/20_gui/cpp-gui-template-sdl2/build/debug2/src/app && cd /D D:\20_gui\cpp-gui-template-sdl2\build\debug2\src\app && D:\cmake-3.28.3\bin\cmake.exe -E copy_directory D:/20_gui/cpp-gui-template-sdl2/src/assets D:/20_gui/cpp-gui-template-sdl2/build/debug2/src/app/../share""
D:\w64devkit\bin/ld.exe: cannot find /subsystem:windows: No such file or directory
D:\w64devkit\bin/ld.exe: cannot find /entry:mainCRTStartup: No such file or directory
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

=========================

Any suggestion for this error? 
thanks

**Comment by MartinHelmut** [comment]
Hey @pingwuu, thanks for reporting a detailed issue. I assume with **w64devkit** you mean this: https://github.com/skeeto/w64devkit ? Searching a bit about this based on your error the problem seems to be the custom entry point the template defines in: https://github.com/MartinHelmut/cpp-gui-template-sdl2/blob/main/src/app/cmake/packaging/Windows.cmake#L2 , setting `subsystem:windows` and `entry:mainCRTStartup`. Maybe a related issue I found is this: https://github.com/skeeto/w64devkit/issues/21

To not just assume, I will need to reproduce and test the problem further. Thankfully, if it is the mentioned Docker file, this should be fairly straight forward. Give me some time and I will update this issue with what I can find out, best case I can update the repo itself ✌🏻

**Comment by pingwuu** [comment]
> Hey @pingwuu, thanks for reporting a detailed issue. I assume with **w64devkit** you mean this: https://github.com/skeeto/w64devkit ? 
===
Yes

**Comment by MartinHelmut** [comment]
Perfect 🙌🏻 I'll have a look today/this weekend and let you know :)

**Comment by MartinHelmut** [comment]
I was able to solve the issue. It is not yet in main, but you can have a look at the PR with the changes: https://github.com/MartinHelmut/cpp-gui-template-sdl2/pull/20 I will test this a bit more before merge, but you should be good to go.

TL;DR:

In `src/app/cmake/packaging/Windows.cmake`:

```diff
- set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /subsystem:windows /entry:mainCRTStartup")
+ if (MINGW)
+   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-subsystem,windows,-Bstatic,--whole-archive -lwinpthread -Wl,--no-whole-archive")
+   set(CMAKE_CXX_STANDARD_LIBRARIES "-static-libgcc -static-libstdc++")
+ else ()
+   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /subsystem:windows /entry:mainCRTStartup")
+ endif ()
```

**Comment by MartinHelmut** [comment]
Merged the PR. Let me know if all works now for you :)

**Comment by pingwuu** [comment]
Great, now buld and run worked well, thanks for your support quickly

**Comment by MartinHelmut** [comment]
No problem, happy it works for you!

One last thing to note though: When testing I found out that the SDL function `SDL_GetDisplayDPI` returns different DPI values when building with MinGW on **high DPI** displays. In case this is relevant for you. I have no good fix yet and even think about upgrading the template to SDL3 for better high DPI handling. Not sure yet. But so you're aware.

**Comment by pingwuu** [comment]
Hi @MartinHelmut , there is same issue on [Build failed with w64devkit on Windows Platform #1](https://github.com/MartinHelmut/cpp-small-gui-project-sdl2/issues/1), and the failures also occured on depend repos, would you help have a look?
